src/components/hero/interactive.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+
+import { NotificationManager } from "react-notifications";
+import { serveFile } from "./utils";
+import {
+  InputFileSpecification,
+  ColumnSpecification,
+  BackButton,
+  SubmitButton,
+} from "./subcomponents";
+
+interface InteractiveTerminalProps {
+  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+}
+
+interface InteractiveTerminalState {
+  column_id: string;
+  column_value: string;
+  column_sort: string;
+  column_kind: string;
+  data_file: string;
+  loading: boolean;
+}
+
+export default class InteractiveTerminal extends React.Component<
+  InteractiveTerminalProps,
+  InteractiveTerminalState
+> {
+  constructor(props) {
+    super(props);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.setState({
+      column_id: "",
+      column_value: "",
+      column_sort: "",
+      column_kind: "",
+      data_file: "",
+      loading: false,
+    });
+  }
```

Comment:
You can directly set `state` as a class member instead of using the `componentWillMount` callback.